frontend/src/views/ElectionDetailPage.vue

```diff
@@ -1,46 +1,16 @@
 <template>
-  <div v-if="election !== null">
-    <h1>{{ election.title }}</h1>
-    <p>{{ election.description }}</p>
-    <p>A total of {{ election.questions.length }} questions</p>
-    <p>and {{ election.questions.flatMap(q => q.answers).length }} answers</p>
-
-    <div v-if="election.is_owned">
-      <template v-if="election.state == 'CREATED'">
-        <h4>Election is not yet opened, you can edit or open it</h4>
-        <p><button v-on:click="editElection()">Edit election</button></p>
-        <p><button v-on:click="openElection()">Open election</button></p>
-      </template>
-      <template v-else-if="election.state == 'OPENED'">
-        <h4>Election is opened</h4>
-        <p><button v-on:click="voteInElection()">Vote in this election!</button></p>
-        <p><button v-on:click="closeElection()">Close election and count votes</button></p>
-      </template>
-      <template v-else-if="election.state == 'CLOSED'">
-        <h4>Election is closed</h4>
-        <p><button v-on:click="showElectionResults()">See election results</button></p>
-      </template>
-    </div>
-    <div v-else>
-      <template v-if="election.state == 'CREATED'">
-        <h4>Election is not yet opened</h4>
-      </template>
-      <template v-else-if="election.state == 'OPENED'">
-        <p><button v-on:click="voteInElection()">Vote in this election!</button></p>
-      </template>
-      <template v-else-if="election.state == 'CLOSED'">
-        <h4>Election is closed</h4>
-        <p><button v-on:click="showElectionResults()">See election results</button></p>
-      </template>
-    </div>
-  </div>
+  <election-detail v-if="election" :election="election"></election-detail>
```

Comment:
this could be self-closing component https://vuejs.org/v2/style-guide/#Self-closing-components-strongly-recommended